### PR TITLE
Refactor sort command

### PR DESCRIPTION
**Pull request by wardetu**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java [hunk 1]
```diff
@@ -13,17 +13,13 @@
...
-            if (sortOrder.equalsIgnoreCase("name")) {
-                return int1.getName().compareTo(int2.getName());
-            } else {
-                return int1.getFrom().compareTo(int2.getFrom());
-            }
-        };
+        Comparator<Internship> baseComparator =
+                sortOrder.equalsIgnoreCase("name")
+                        ? Comparator.comparing(Internship::getName)
+                        : Comparator.comparing(Internship::getFrom);
```
Comment (duongphammmm) [inline src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java:22]: do we need to check for invalid sortOrder here? eg. not "name" and also not "time"? or can we assume that parser already handles this?

**Review by duongphammmm — COMMENTED** [review]
Overall very neat changes!

File: src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java (hunk 1 above)
Comment (wardetu) [reply src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java:22]: The parser would have handled it.

Comment (helloImHai) [reply src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java:22]: I think in parser it was quite safe so sortComparator doesn't have to check iirc, @wardetu please reassure

**Review by helloImHai — APPROVED** [review]
=
Clean and I love it 🛀🛀🛀

File: src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java (hunk 1 above)
Comment (helloImHai) [reply src/main/java/seedu/address/logic/commands/sort/SortInternshipsCommand.java:22]: Oh oops you did reassure lol mb my wifi

**Pull request merged**